obspy/io/sac/sactrace.py

```diff
@@ -1158,10 +1158,14 @@ def read(cls, source, headonly=False, ascii=False, byteorder=None,
                                             checksize=checksize)
         if not debug_strings:
             for i, val in enumerate(hs):
-                val = _ut._clean_str(val, strip_whitespace=False)
+                if 'encode' in kwargs.keys():
+                    val = _ut._clean_str(val, encode=kwargs['encode'],
+                                         strip_whitespace=False)
+                else:
+                    val = _ut._clean_str(val, strip_whitespace=False)
                 if val.startswith(native_str('-12345')):
                     val = HD.SNULL
-                hs[i] = val
+                hs[i] = val.encode('ascii', 'replace')
```

Comment:
I can't follow this logic, why always encode to ASCII when the user might have specified a different encoding?